### ISSUE TITLE
Connecting to DataStax Astra

### ISSUE THREAD

**Issue by phact** [apache/cassandra-gocql-driver#1452]
Currently connecting to DataStax Astra requires the secure connect bundle to be extracted and the certs to be added individually in the connection string like so:

https://gist.github.com/phact/b6719203529d9322b50a319b41011d72

Other c* drivers allow it in a single line:

https://docs.datastax.com/en/astra/aws/doc/dscloud/astra/dscloudConnectJavaDriver.html

Would you accept a PR to support this functionality?

**Comment by jorgebay** [comment]
I think the code you shared will allow connecting to a single node, as the rest of the nodes have different server names.

**Comment by mpenick** [comment]
Internally, the other connections would need to connect using  `metadata.ContactInfo.SniProxyAddress` + the `host_id` (from the system tables) as the SNI servername.

**Comment by martin-sucha** [comment]
Related: #1487

**Comment by martin-sucha** [comment]
Once https://github.com/gocql/gocql/pull/1629 is merged, it should allow you to use custom TLS connection setup, so you could use `host_id` in the SNI server name.

**Comment by OleksiienkoMykyta** [comment]
Looks like (https://github.com/apache/cassandra-gocql-driver/pull/1629) helps with it. @phact have you had a chance to test it after the mentioned PR landed?
Also, there are [gocql-astra](https://github.com/datastax/gocql-astra) which specified for Astra

**Comment by joao-r-reis** [comment]
Yes, there's also #1681 that I'd like to work on in the near future but it's not top priority right now.
